src/cloudant/indexes.py

```diff
@@ -0,0 +1,271 @@
+    be used by the database convenience methods
+    :func:`~cloudant.database.CloudantDatabase.create_index`,
+    :func:`~cloudant.database.CloudantDatabase.delete_index`, and
+    :func:`~cloudant.database.CloudantDatabase.get_all_indexes`.  It is
+    recommended that you use those methods to manage an index rather than
+    directly interfacing with SearchIndex objects.
+
+    :param CloudantDatabase database: A Cloudant database instance used by the
+        SearchIndex.
+    :param str design_document_id: Optional identifier of the design document.
+    :param str name: Optional name of the index.
+    :param **kwargs: Options used to construct the index definition for the
+        purposes of index creation.  For more details on valid options See
+        :func:`~cloudant.database.CloudantDatabase.create_index`.
+    """
+    def __init__(self, database, design_document_id=None, name=None, **kwargs):
+        super(SearchIndex, self).__init__(
+            database,
+            design_document_id,
+            name,
+            **kwargs
+        )
+        self._type = 'text'
+
+    def _def_check(self):
+        """
+        Checks that the definition provided contains only valid arguments for a
+        text index.
+        """
+        valid_args = {
+            'fields': list,
+            'default_field': dict,
+            'selector': dict
+        }
+        if self._def != {}:
+            for key, val in self._def.iteritems():
+                if key not in valid_args.keys():
+                    msg = 'Invalid argument: {0}'.format(key)
+                    raise CloudantArgumentError(msg)
+                if not isinstance(val, valid_args[key]):
+                    msg = (
+                        'Argument {0} is not an instance of expected type: {1}'
+                    ).format(key, valid_args[key])
+                    raise CloudantArgumentError(msg)
+
+class SpecialIndex(Index):
+    """
+    Provides an interface for viewing the "special" primary index of a database.
+    Primarily meant to be used by the database convenience method
+    :func:`~cloudant.database.CloudantDatabase.get_all_indexes`.  It is
+    recommended that you use that method to view the special index rather than
+    directly interfacing with the SpecialIndex object.
+
+    :param CloudantDatabase database: A Cloudant database instance used by the
+        SpecialIndex.
+    :param **kwargs: Only an ascending order _id field parameter is allowed as
+        an argument for a SpecialIndex.
+        For example ``fields=[{'_id': 'asc'}]``.
+    """
+    def __init__(self, database, **kwargs):
+        if kwargs.keys() != ['fields']:
+            msg = (
+                '{0} provided as argument(s).  The \"special\" index only '
+                'contains the \"fields\" argument.'
+            ).format(kwargs)
+            raise CloudantArgumentError(msg)
+        if kwargs['fields'] != [{'_id': 'asc'}]:
```

Comment:
After reading your comment and giving some more thought to the actual purpose of this class, I decided to refactor it slightly.  Essentially the purpose of a SpecialIndex in this library is for use by the database API method `get_all_indexes()`.  'get_all_indexes()` retrieves the indexes in the database through the _index endpoint and wraps the primary "special" index as a SpecialIndex object as well as wrapping JSON indexes as Index objects and TEXT indexes as SearchIndex objects and returns the index list to the caller.

Since you cannot really do anything with a SpecialIndex object other than instanciating it, it didn't really make much sense to have any kind of validation as was originally being done.  So the validation was removed and the method signature was changed to match that if the parent Index class.  This change provides us with flexibility if we (Cloudant) decide to change the "special" index definition server side at any point down the road (unlikely but still possible), this library implementation will still be able to handle it.

This is done in daf19a6dd600acb796fb87dc7f78a0431b5940dc and 5740aa251ca744ad62d53bb36986bf68e2da4340.